pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/JmxMetricsIntegrationTest.java

```diff
@@ -138,6 +153,39 @@ public void testMultiStageMigrationMetric() throws Exception {
     assertEquals((Long) MBEAN_SERVER.getAttribute(multiStageMigrationMetric, "Count"), 6L);
   }
 
+  @Test
+  public void testEstimatedMseServerThreadsBrokerMetric() throws Exception {
+    ObjectName estimatedMseServerThreadsMetric = new ObjectName(PINOT_JMX_METRICS_DOMAIN,
+        new Hashtable<>(Map.of("type", BROKER_METRICS_TYPE,
+            "name", "\"pinot.broker.estimatedMseServerThreads\"")));
+
+    assertEquals((Long) MBEAN_SERVER.getAttribute(estimatedMseServerThreadsMetric, "Value"), 0L);
+
+    ExecutorService executorService = Executors.newSingleThreadExecutor();
+
+    executorService.submit(() -> {
+      while (true) {
+        try {
+          postQuery(
+              "SET useMultiStageEngine=true; SELECT AirlineID, AVG(ArrDelay) FILTER (WHERE ArrDelay > 0) FROM mytable"
+                  + " GROUP BY AirlineID;");
```

Comment:
Nit: You can use the sleep function to be sure the query keeps running for a while, increasing the chance of finding a thread running in the check below.